ledger-service/http-json/src/main/scala/com/digitalasset/http/DbStartupResult.scala

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.http
+
+import com.daml.http.dbbackend.ContractDao
+import com.daml.logging.{ContextualizedLogger, LoggingContextOf}
+import cats.effect.IO
+import com.daml.http.util.Logging.InstanceUUID
+
+sealed trait DbStartupResult
+object DbStartupResult {
+  case object GracefullyExit extends DbStartupResult
+  case object Continue extends DbStartupResult
+
+  private def fromBool(shouldGracefullyExit: Boolean): DbStartupResult =
+    if (shouldGracefullyExit) GracefullyExit else Continue
+
+  private[this] val logger = ContextualizedLogger.get(getClass)
+
+  def fromStartupMode(dao: ContractDao, dbStartupMode: DbStartupMode)(implicit
```

Comment:
This deserves a comment on the meaning of `None` and the difference between successfully returning `None` and throwing an exception. Looking at the code, it seems the difference is fairly arbitrary and the callsite handles both more or less the same so you could also consider if you really need to represent those two types of errors differently.